armory/data/adversarial/resisc_adversarial_patch.py

```diff
@@ -0,0 +1,82 @@
+"""
+TensorFlow Dataset for resisc45 attacked by Adversarial Patch with adv/clean splits
+"""
+
+import tensorflow.compat.v2 as tf
+import tensorflow_datasets.public_api as tfds
+
+_DESCRIPTION = """\
+NWPU-RESISC45 adversarial image dataset for DenseNet121
+DenseNet121 training on benign images: 
+    architecture name, batch_size, initial learning rate, optimizer configuration, loss, metrics
+
+AdversarialPatch
+    max_iter 1000
+    scale_min, scale_max 0.4
+    learning_rate 2.0
+    (By definition targeted)
+"""
+
+_URL = "https://armory-public-data.s3.us-east-2.amazonaws.com/imagenet-adv/ILSVRC12_ResNet50_PGD_adversarial_dataset_v1.0.tfrecords"  # to be updated
+_PATH = "armory/data/data/adversarial/RESISC45_DenseNet121_AdvPatch_adversarial_dataset_v0.1.tfrecords"
+
+
+class ImagenetAdversarial(tfds.core.GeneratorBasedBuilder):
+    """ RESISC45_DenseNet121_AdvPatch_adversarial_dataset_v0.1.tfrecords """
+
+    VERSION = tfds.core.Version("1.0.0")
+
+    def _info(self):
+        return tfds.core.DatasetInfo(
+            builder=self,
+            description=_DESCRIPTION,
+            features=tfds.features.FeaturesDict(
+                {
+                    "image": tfds.features.Tensor(shape=[224, 224, 3], dtype=tf.uint8),
+                    "label": tfds.features.Tensor(shape=(), dtype=tf.int64),
+                }
+            ),
+            supervised_keys=("image", "label"),
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        path = _PATH  # dl_manager.download_and_extract(_URL)
+        splits = [
+            tfds.core.SplitGenerator(name=x, gen_kwargs={"path": path, "split": x})
+            for x in ("adversarial", "clean")
```

Comment:
What's the motivation for having a "clean" split for this? We already have a clean version of resisc45. I would prefer not to have the user have two large duplicative datasets.

Also, how would you split this into train and test? Is it already split? If so, that is unclear.

Reply:
The "clean" version makes sense if it is otherwise difficult to pair with the "adversarial" split.

However, we do need to figure out train/test splits in this context, so that we're not returning adversarial test results on the training set.

Reply:
This is something that we've been discussing internally on how to best address.  Obviously, including the clean examples with every adversarial dataset is doubles the size of the dataset, which could be prohibitive to storage and downloading.

The other way to address it is to deliver adversarial datasets with only adversarial examples.  If examples for all datasets (clean and adv) are tagged in the tfds with a unique name, which I think is already done for many, if not all, datasets, and then we can do a brute-force pairing of adversarial and clean examples (this would take N^2 time), but it's probably okay since this doesn't need to be done often.